src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -420,6 +429,32 @@ private void initiateOtpFlow(AuthorizeRequest request,
         model.addAttribute("authorizeCommand", request);
     }
 
+    /**
+     * @should return error page view if OAuth2 details are missing
+     * @should populate authorizeCommand
+     */
+    @GetMapping("/verification")
+    public String verificationView(@ModelAttribute("authorizeCommand") VerificationRequest request,
+                            BindingResult bindingResult, Model model) {
+        if (StringUtils.isEmpty(request.getClient_id()) || StringUtils.isEmpty(request.getRedirect_uri())) {
+            model.addAttribute(ERROR_MSG, "error.page.access.denied");
+            model.addAttribute(ERROR_SUB_MSG, "public.error.page.access.denied.text");
+            return ERRORPAGE_VIEW;
+        }
+
+        model.addAttribute(USERNAME, request.getUsername());
```

Comment:
I think you really need to retain all attributes from the request through this entire process.
This is all fine for OAuth2 but OpenID Connect can have 14 different parameters